lib/ansible/modules/cloud/amazon/ec2_vpn_connection.py

```diff
@@ -0,0 +1,199 @@
+      description: The ID of the customer gateway at your end of the VPN connection.
+      type: string
+      sample: cgw-abcdefgh
+'''
+
+try:
+    from botocore.exceptions import ClientError
+    HAS_BOTOCORE = True
+except ImportError:
+    HAS_BOTOCORE = False
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (boto3_conn, camel_dict_to_snake_dict,
+    ec2_argument_spec, get_aws_connection_info)
+
+
+def create_vpn_connection(module, ec2):
+    filters=[
+        {'Name': 'state', 'Values': ['pending', 'available']},
+        {'Name': 'customer-gateway-id', 'Values': [module.params.get('customer_gateway_id')]},
+        {'Name': 'vpn-gateway-id', 'Values': [module.params.get('vpn_gateway_id')]},
+    ]
+
+    if module.params.get('static_routes_only'):
+        filters.append({'Name': 'option.static-routes-only', 'Values': ['true']})
+
+    vpns = ec2.describe_vpn_connections(Filters=filters)
+
+    if vpns['VpnConnections']:
+        # only one will ever be returned for a (customer-gateway-id, vpn-gateway-id) pair
+        return (camel_dict_to_snake_dict(vpns['VpnConnections'][0]), False)
+    else:
+        vpn = ec2.create_vpn_connection(
+            DryRun=False,
+            Type='ipsec.1',
+            CustomerGatewayId=module.params.get('customer_gateway_id'),
+            VpnGatewayId=module.params.get('vpn_gateway_id'),
+            Options={
+                'StaticRoutesOnly': module.params.get('static_routes_only'),
+            },
+        )
+
+        return (camel_dict_to_snake_dict(vpn['VpnConnection']), True)
+
+
+def delete_vpn_connection(module, ec2):
+    vpns = ec2.describe_vpn_connections(
+        VpnConnectionIds=[module.params.get('vpn_connection_id')],
+        Filters=[
+            {'Name': 'state', 'Values': ['pending','available']},
+        ],
+    )
+
+    deleted_ids = []
+
+    for vpn in vpns['VpnConnections']:
+        ec2.delete_vpn_connection(VpnConnectionId=vpn['VpnConnectionId'])
+        deleted_ids.append(vpn['VpnConnectionId'])
+
+    return (deleted_ids)
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            state=dict(default='present', choices=['present', 'absent']),
+            vpn_connection_id=dict(required=False),
+            customer_gateway_id=dict(required=False),
+            vpn_gateway_id=dict(required=False),
+            static_routes_only=dict(required=False,default=False,type='bool'),
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ('state', 'present', ['customer_gateway_id', 'vpn_gateway_id']),
+            ('state', 'absent', ['vpn_connection_id']),
+        ],
+    )
+
+    if not HAS_BOTOCORE:
+        module.fail_json(msg='botocore is required.')
+
+    if not HAS_BOTO3:
```

Comment:
Can you check for `boto3` first, since it requires botocore? That way users who have neither will get both by installing `boto3`.